### PR TITLE
Support for @recommended:languages search

### PR DESCRIPTION
**Pull request by digitarald**
Similar to keymaps, this adds the same mechanism for languages.

### CODE REVIEW

File: src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts [hunk 1]
```diff
@@ -501,7 +504,10 @@ class ExtensionsContributions extends Disposable implements IWorkbenchContributi
 				id: MenuId.CommandPalette,
 				when: CONTEXT_HAS_GALLERY
 			},
-			run: () => runAction(this.instantiationService.createInstance(SearchExtensionsAction, '@category:"programming languages" @sort:installs '))
+			run: async () => {
+				const recommended = await this.tasExperimentService.getTreatment<boolean>('recommendedLanguages');
```
Comment (lszomoru) [inline src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts:508]: tasExperimentService is marked as an optional dependency so before using it, you should check that its different than `undefined`.

File: src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts [hunk 2]
```diff
@@ -388,6 +389,7 @@ class ExtensionsContributions extends Disposable implements IWorkbenchContributi
 		@IInstantiationService private readonly instantiationService: IInstantiationService,
 		@IDialogService private readonly dialogService: IDialogService,
 		@ICommandService private readonly commandService: ICommandService,
+		@optional(ITASExperimentService) private readonly tasExperimentService: ITASExperimentService,
```
Comment (lszomoru) [inline src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts:392]: The pattern that we are using consistently across the codebase is that:
* This should not be declared as `private readonly`
* Instead declare a private field `private tasExperimentService: ITASExperimentService | undefined;`

**Review by lszomoru — APPROVED** [review]
Given that the comments are addressed, this look good to me.

File: src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts (hunk 2 above)
Comment (digitarald) [reply src/vs/workbench/contrib/extensions/browser/extensions.contribution.ts:392]: Ran with `private tasExperimentService?: ITASExperimentService;`

**Pull request merged**